mlrun/projects/project.py

```diff
@@ -3761,6 +3767,79 @@ def _validate_file_path(self, file_path: str, param_name: str):
+        project_producer_tag=None,
+    ) -> typing.Optional[ArtifactProducer]:
+        """
+        Resolve the artifact producer of the given artifact.
+        If the artifact producer is a project, the artifact is registered with the current project as the producer.
+        If not, the artifact is registered with the original producer. If the artifact is already registered, the
+        artifact is not registered again.
+
+        :param artifact:                The artifact to resolve its producer.
+        :param project_producer_tag:    The tag to use for the project as the producer. If not provided, a tag will be
+        generated for the project.
+        :return:                        A tuple of the resolved producer and the resolved artifact.
+        """
+
+        if not isinstance(artifact, str) and artifact.producer:
+            # if the artifact was imported from a yaml file, the producer can be a dict
+            if isinstance(artifact.spec.producer, ArtifactProducer):
+                producer_dict = artifact.spec.producer.get_meta()
+            else:
+                producer_dict = artifact.spec.producer
+
+            if "project" != producer_dict.get("kind", ""):
+                return ArtifactProducer(
+                    name=producer_dict.get("name", ""),
+                    kind=producer_dict.get("kind", ""),
+                    project=producer_dict.get("project", ""),
+                    tag=producer_dict.get("tag", ""),
+                )
+
+        # the artifact is not registered, register it with the project as the producer
+        project_producer_tag = (
+            project_producer_tag or self._get_hexsha() or str(uuid.uuid4())
+        )
+        return ArtifactProducer(
+            "project",
+            self.metadata.name,
+            self.metadata.name,
+            tag=project_producer_tag,
+        )
+
+    def _resolve_existing_artifact(
+        self, item, tag=None
+    ) -> typing.Optional[typing.Union[Artifact, DatasetArtifact, ModelArtifact]]:
+        """
+        Check if there is and existing artifact with the given item and tag.
+        If there is, return the existing artifact. Otherwise, return None.
+
+        :param item:    The item (or key) to check if there is an existing artifact for.
+        :param tag:     The tag to check if there is an existing artifact for.
+        :return:        The existing artifact if there is one, otherwise None.
+        """
+        try:
+            if isinstance(item, str):
+                existing_artifact = self.get_artifact(key=item, tag=tag)
+            else:
+                existing_artifact = self.get_artifact(
+                    key=item.key,
+                    tag=item.tag,
+                    iter=item.iter,
+                    tree=item.tree,
+                )
+            if existing_artifact is not None:
+                return existing_artifact.from_dict(existing_artifact)
```

Comment:
Is it even possible to get `None` here?

Reply:
It is now, see my comment in https://github.com/mlrun/mlrun/pull/5283/files#diff-c972cfb48585c0ee85184b83808a9bf592a0ddd60a07cba789023649354f6cb8R3338-R3339